pylint/checkers/strings.py

```diff
@@ -476,7 +476,9 @@ def _check_new_format_specifiers(self, node, fields, named):
                         break
                 else:
                     warn_error = False
-                    if hasattr(previous, 'getitem'):
+                    if isinstance(previous, astroid.ClassDef):
```

Comment:
Why not letting the ``getitem`` call do its job? This can work too:

```
>>> class A(type):
...     def __getitem__(self, obj):
...         return obj
... 
>>> class B(object):
...    __metaclass__ = A
... 
>>> "{}".format(B)
"<class '__main__.B'>"
>>> "{0['test']}".format(B)
'test'
>>> 
```